packages/analytics-browser/src/browser-client.ts

```diff
@@ -78,6 +78,8 @@ export class AmplitudeBrowser extends AmplitudeCore implements BrowserClient {
     // Session ID is handled differently than device ID and user ID due to session events
     this.setSessionId(options.sessionId ?? this.config.sessionId ?? Date.now());
 
+    await super._init(this.config);
```

Comment:
Question: 
In the setSessionId function, we also sent the session event. 
Since we set the config after the setSessionId, will the session event have the right configuration?

Reply:
Yes, the session event should have the right configuration as we have `this.config = browserOptions` before `setSessionId()`.

Reply:
We reset the timeline plugin in the super._init functions. Does that mean there might be some remaining plugins overwrite the session event?

Reply:
Yes, session Id can be reset in `plugin.teardown()` as it's called in `super._init()`. But technically, `teardown()` is designed for releasing resources, etc. Users should not change configuration there. Current implementation also doesn't guarantee that configuration is unchanged in `teardown()`. In current implementation, if session Id is changed in `teardown()`, the session Id of session end event will be wrong.

Reply:
Thanks for checking this.